### DISCUSSION TITLE
Rowid on view

### DISCUSSION THREAD

**Discussion by florentfougeres** [duckdb/duckdb#13869] [Q&A]
It's not possible to make the rowid pseudo-column on a view

```sql
select rowid, * from view_point;
```

### Error
```log
Binder Error: Referenced column "rowid" not found in FROM clause!
Candidate bindings: "view_point.id"
LINE 1: select rowid, * from view_point;
```

Is there an alternative or a workaround? I need to have an incremental column on a view.

**Comment by florentfougeres** [comment]
I found 

`ROW_NUMBER() OVER () AS row_number`

But is it the best idea?

**Comment by hbruch** [comment]
Order is preserved in some cases but not all, see https://duckdb.org/docs/stable/sql/dialect/order_preservation.html.
I.e. for views with _multiple joined tables_ it is not.
